File: JAVA/edu/hit/ir/ltpService/LTPService.java
package edu.hit.ir.ltpService;

import java.io.IOException;

import org.jdom.JDOMException;

import edu.hit.ir.ltpService.CirService;

public class LTPService {
	CirService cs;

	public LTPService(String authorization) {
		super();
		this.cs = new CirService(authorization);
	}

	public LTML analyze(String option, String analyzeString) throws JDOMException, IOException{
		cs.setAnalysisOptions(option);
		cs.setXmlOption(false);
		LTML ltml = new LTML();
		ltml.setEncoding(cs.getEncoding());
		ltml.build(cs.Connect(analyzeString));
		ltml.setOver();
		return ltml;
	}
	
	public LTML analyze(String option, LTML ltmlIn) throws JDOMException, IOException{
		cs.setAnalysisOptions(option);
		cs.setXmlOption(true);
		LTML ltml = new LTML();
		ltml.setEncoding(cs.getEncoding());
		ltml.build(cs.Connect(ltmlIn.getXMLStr()));
		ltml.setOver();
		return ltml;
	}
	
	public void setAnalysisOptions(String op) {
		cs.setAnalysisOptions(op);
	}

	public void setAuthorization(String authorization) {
		cs.setAuthorization(authorization);
	}

	public boolean setEncoding(String en) {
		return cs.setEncoding(en);
	}

	public String getEncoding() {
		return cs.getEncoding();
	}
	
}

File: C++/__ltpService/LTPService.cpp
#include "LTPService.h"

namespace HIT_IR_LTP{
	LTPService::~LTPService(void)
	{
	}

	void LTPService::Analyze(const std::string& option, const std::string& analyzeString, LTML& ltml_out)
	{
		ls.SetAnalysisOptions(option);
		ls.SetXmlOption(false);
		std::string resultStr;
		if(analyzeString.size() == 0)
		{
			std::cerr<<"Input analyzeString is null!"<<std::endl;
		}
		else
		{
			//		printf("analyzeString: \n%s\n", analyzeString.c_str());
			ls.Analyze(analyzeString, resultStr);
			//		printf("result: \n%s\n", resultStr.c_str());
			ltml_out.ClearDOM();
			ltml_out.SetEncoding(ls.GetEncoding());
			ltml_out.LoadLtml(resultStr);
			ltml_out.SetOver();
		}
	}

	void LTPService::Analyze(const std::string& option, const LTML& ltml_in, LTML& ltml_out)
	{
		ls.SetAnalysisOptions(option);
		ls.SetXmlOption(true);
		std::string resultStr;
		ls.Analyze(ltml_in.GetXMLStr(), resultStr);
		ltml_out.ClearDOM();
		ltml_out.SetEncoding(ls.GetEncoding());
		ltml_out.LoadLtml(resultStr);
		ltml_out.SetOver();
	}
}
